packages/core/src/domain/session/sessionStoreOperations.ts

```diff
@@ -0,0 +1,107 @@
+import { setTimeout } from '../../tools/timer'
+import { dateNow } from '../../tools/utils/timeUtils'
+import { generateUUID } from '../../tools/utils/stringUtils'
+import { isChromium } from '../../tools/utils/browserDetection'
+import { SESSION_EXPIRATION_DELAY } from './sessionConstants'
+import type { SessionState, SessionStore } from './sessionStore'
+import { isSessionInExpiredState } from './sessionStore'
+
+type Operations = {
+  process: (sessionState: SessionState) => SessionState | undefined
+  after?: (sessionState: SessionState) => void
+}
+
+export const LOCK_RETRY_DELAY = 10
+export const LOCK_MAX_TRIES = 100
+const bufferedOperations: Operations[] = []
+let ongoingOperations: Operations | undefined
+
+export function processSessionStoreOperations(operations: Operations, sessionStore: SessionStore, numberOfRetries = 0) {
+  const { retrieveSession, persistSession, clearSession } = sessionStore
+  const lockEnabled = isLockEnabled()
+
+  if (!ongoingOperations) {
+    ongoingOperations = operations
+  }
+  if (operations !== ongoingOperations) {
+    bufferedOperations.push(operations)
+    return
+  }
+  if (lockEnabled && numberOfRetries >= LOCK_MAX_TRIES) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏is it needed?
```suggestion
  if (numberOfRetries >= LOCK_MAX_TRIES) {
```

Reply:
Somehow yes - If `lockEnabled` is removed - the code works only because LOCK_MAX_TRIES > 0. I'd prefer not to check the lock retry mechanism when locks are not enabled.

Reply:
Setting `LOCK_MAX_TRIES` to 0 seems unlikely but I get the intent.
Could we make a test to enforce that?

Reply:
👍